executor/runcexecutor/executor.go

```diff
@@ -184,32 +211,140 @@ func (w *runcExecutor) Exec(ctx context.Context, meta executor.Meta, root cache.
 		return err
 	}
 
-	logrus.Debugf("> running %s %v", id, meta.Args)
+	forwardIO, err := newForwardIO(stdin, stdout, stderr)
+	if err != nil {
+		return errors.Wrap(err, "creating new forwarding IO")
+	}
+	defer forwardIO.Close()
+
+	pidFilePath := filepath.Join(w.root, "runc_pid_"+identity.NewID())
+	defer os.RemoveAll(pidFilePath)
 
-	status, err := w.runc.Run(ctx, id, bundle, &runc.CreateOpts{
-		IO: &forwardIO{stdin: stdin, stdout: stdout, stderr: stderr},
+	logrus.Debugf("> creating %s %v", id, meta.Args)
+	err = w.runc.Create(ctx, id, bundle, &runc.CreateOpts{
+		PidFile: pidFilePath,
+		IO:      forwardIO,
 	})
-	logrus.Debugf("< completed %s %v %v", id, status, err)
-	if status != 0 {
+	if err != nil {
+		return err
+	}
+	forwardIO.release()
+
+	defer func() {
+		go func() {
```

Comment:
why goroutine?

Reply:
To return the result back to the solver faster. As this pr introduces new process invocations that were not needed before and therefore makes running a container slower.